pkg/storaged/btrfs/btrfs-tool.py

```diff
@@ -241,15 +243,34 @@ def cmd_poll(opt_mount):
     sys.stdout.flush()
 
 
+def unshare_mounts():
+    # The os.unshare function is available since Python 3.12, but we
+    # still need to support Python 3.11.
+    libc = ctypes.CDLL(None)
+    libc.unshare.argtypes = [ctypes.c_int]
+    ret = libc.unshare(2**17)
+    if ret != 0:
+        raise Exception("unshare(CLONE_NEWNS) failed")
+
+
 def cmd_do(uuid, cmd):
     debug(f"DO {uuid} {cmd}")
-    with mount_database() as db:
-        filesystems = list_filesystems()
-        for fs in filesystems.values():
-            if fs['uuid'] == uuid:
-                mp = force_mount_point(db, fs, opt_repair=True)
-                with context_chdir(mp):
-                    subprocess.check_call(cmd)
+    filesystems = list_filesystems()
+    for fs in filesystems.values():
+        if fs['uuid'] == uuid:
+            path = tempfile.mkdtemp()
+            dev = fs['devices'][0]
+            try:
+                unshare_mounts()
+                subprocess.check_call(["mount", "--make-rprivate", "/"])
+                subprocess.check_call(["mount", dev, path])
+                try:
+                    with context_chdir(path):
+                        subprocess.check_call(cmd)
```

Comment:
check_call has a `cwd` argument, you don't need the context_chdir bit.

Reply:
Ah, done.